library/asb_set_state.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/python
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.0',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+
+import json
+import os
+
+from ansible.module_utils.basic import AnsibleModule
+try:
+    from kubernetes import client, config
+    from kubernetes.client.rest import ApiException
+    config.load_kube_config()
+    api = client.CoreV1Api()
+except Exception as error:
+    ansible_module.fail_json(
+        msg="Error attempting to load kubernetes client: {}".format(error))
+
+
+ENV_NAME = "POD_NAME"
+ENV_NAMESPACE = "POD_NAMESPACE"
+
+
+def create_config_map(name, namespace, data):
+    meta = client.V1ObjectMeta(name=name)
+    body = client.V1ConfigMap(metadata=meta, data=data)
+    try:
+        res = api.create_namespaced_config_map(namespace, body)
+    except ApiException as e:
+        module.fail_json(
+            msg="Error attempting to create namespaced configmap: {}".format(e))
+
+
+def update_config_map(name, namespace, data):
+    body = client.V1ConfigMap(data=data)
+    try:
+        res = api.patch_namespaced_config_map(name, namespace, body)
+    except ApiException as e:
+        module.fail_json(
+            msg="Error attempting to update configmap: {}".format(e))
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            key=dict(required=True, type='str'),
+            value=dict(required=True, type='str'),
+        ),
```

Comment:
Would it not also need a list of keys?
Are you thinking the normal usecase here would be to do all the work and then as the last action in the APB set all the state?

Reply:
Yeah at the time I was thinking it might look like:
```asb_set_state:
         args:
             [ {"key": "xyz", "value": "123"}, {"key": "abc", "value": "123"}]
```
or maybe even just a dict as arg 

I was just thinking it's less API calls however you are much more familiar with the use case than I am. I think the above is probably ugly and over complex so it probably doesn't make sense. Prefer to call with `with_items` if needs be

Reply:
I think keeping a single key value is ok for now. When thinking about this feature, I imagined a developer creating some state and then really only doing one set state call with a single key. Then getting this back during another action.

Reply:
@maleck13 I tend to agree with @PhilipGough. I think this should replicate the `asb_encode_binding` module where you can supply a list of key/value pairs. I would like to see this module replace all uses of that module that aren't actually providing bind creds.

Reply:
@dymurray thinking about this further I wonder if accepting a list makes sense? From a user perspective are they likely to set all state in one call or make multiple calls to this module when they have some state they would like to set. I'm ok with making the change bust just want to confirm how the inputs to this module should look before doing so? 

@fabianvf any thoughts?

Reply:
It doesn't seem like you gain much by allowing lists, since `with_items` would let you use a list in there anyway, but I don't have any opposition to adding it in if it isn't too hard. We could also just make it freeform and allow people to do

```yaml
- asb_set_state:
    key1: value1
    key2: value2
    ...
    keyN: valueN
```